### DISCUSSION TITLE
DB Documentation

### DISCUSSION THREAD

**Discussion by vdmitriyev** [dmarro89/dare-db#26] [Ideas]
For now documentation of the project is conducted in multiple. Primary README.md is used. There are also further important information hidden in various discussion threads.

The current discussion should be used to discuss further possible improvements and enhancements.

**Comment by vdmitriyev** [comment]
List of important discussion to be added later on into documentation 

* https://github.com/dmarro89/dare-db/discussions/8
* https://github.com/dmarro89/dare-db/discussions/12
* https://github.com/dmarro89/dare-db/discussions/16

**Comment by vdmitriyev** [comment]
Concept/ideas how to handle documentation of the project:

* Documentation should be written for end users and adaptors
* README.md should contain only basic information and a link to a main documentation
* Documentation should be written in markdown  in the folder ```docs```
* Documentation should be automatically build on push through Github Action 
* Documentation should be available as ```gh-pages```

Possible tools to be used (option 1 was selected):

1. Example of the documentation written for [gotify](https://github.com/gotify/server)
    - Documentation example: https://gotify.net/
    - Documentation src: https://github.com/gotify/website
    - Based on docusaurus: https://docusaurus.io/docs
    - Docusaurus: [Docker Deployment](https://docusaurus.community/knowledge/deployment/docker/)
        - Some how doesn't work for now
     

2. Make use of [hugo-geekdoc](https://github.com/thegeeklab/hugo-geekdoc) template for a better UI and UX

3. Make use of [mkdocs](https://www.mkdocs.org/) or [sphinx](https://www.sphinx-doc.org/en/master/) from Python community
    - Example in go project: https://github.com/ko-build/ko/tree/main/docs
    - Requires Python and couple of Python packages
    - Easy to intergrade into VS Code, Github Actions and it is well supported by community

**Reply by vdmitriyev** [reply]
@dmarro89 I had written a small concept on how to proceed with documentation for the project. What would you say?

**Reply by dmarro89** [reply]
> @dmarro89 I had written a small concept on how to proceed with documentation for the project. What would you say?

Thank you for your ideas, I'll analyze them in the next days (I'm very busy in these days).
Anyway it is an important stuff to discuss.

**Reply by vdmitriyev** [reply]
no stress, take your time 👍

**Reply by vdmitriyev** [reply]
@dmarro89 hi

After reviewing 3 options, an example website with documentation has been created.

Documentation is available here: https://vdmitriyev.github.io/daredb-docs/
Sample Github Repo with the docs res is here: https://github.com/vdmitriyev/daredb-docs

If you like it, that you can create the Github Repo with the same name in your personal workspace (```daredb-docs```) and we could migrate provided by me sample Github Repo to your one.

**Reply by dmarro89** [reply]
I'm sorry, I forgot to reply to the thread. It's fine for me.
Thanks for your collaboration!

**Reply by vdmitriyev** [reply]
@dmarro89 you are welcome!

**Reply by dmarro89** [reply]
@vdmitriyev I've forked your documentation repo. 
Very good job, congrats!

Let's add some issues to update it.

**Reply by vdmitriyev** [reply]
@dmarro89 nice! Thank you! 

1. After cloning the repo, you also need to change readme, so it points to the right repository: https://dmarro89.github.io/daredb-docs/ 
2. You also need to configure documentation build and publish process using github actions and settings of the repo

**Reply by dmarro89** [reply]
Configured @vdmitriyev -> https://dmarro89.github.io/daredb-docs/

Thanks. Let's keep this discussion opened for more update about the documentation.

**Reply by vdmitriyev** [reply]
thanks for your efforts, looks good :+1:

**Reply by vdmitriyev** [reply]
I am a bit busy these weeks, but as long as I will have more time, I will get back to discussions / code / docs of DareDB

**Reply by dmarro89** [reply]
Don't worry, no rush.
Take your time 🙂

**Comment by vdmitriyev** [comment]
Initial draft of OpenAPI 3.0 spec
![image](https://github.com/user-attachments/assets/c1ecff14-0815-4e8f-b5c2-b0921b7f5bf2)

**Reply by vdmitriyev** [reply]
@dmarro89 this this what I got so far. Providing a screenshot for now.  But I have also some code that generates web UI based on OpenAPI 3.0 spec

**Reply by dmarro89** [reply]
Thank you very much!
Really good job

**Reply by vdmitriyev** [reply]
Thanks!

**Comment by vdmitriyev** [comment]
PR with the initial OpenAPI 3.0 spec draft - https://github.com/dmarro89/dare-db/pull/61

**Reply by dmarro89** [reply]
Great!
PR Approved

**Reply by vdmitriyev** [reply]
Thank you! Then I am going to merge it

**Reply by vdmitriyev** [reply]
@dmarro89 had to update documentation on OpenAPI 3.0 spec. Could you review and approve it?

**Reply by dmarro89** [reply]
Approved!

**Reply by vdmitriyev** [reply]
Nice. Thanks. It worked. Now also you can view API from OpenAPI 3.0 spec directly using `redocly`: 

https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/dmarro89/dare-db/refs/heads/main/openapi/daredb-openapi_primary.yaml

**Reply by dmarro89** [reply]
Cool!
Really appreciate your contribution ♥️
